packages/statemanager/README.md

```diff
@@ -62,33 +67,63 @@ Have a loot at the extended `CacheOptions` on how to use and leverage the new ca
 The `DefaultStateManager` has a static constructor `fromProof` that accepts one or more [EIP-1186](https://eips.ethereum.org/EIPS/eip-1186) [proofs](./src/stateManager.ts) and will instantiate a `DefaultStateManager` with a partial trie containing the state provided by the proof(s). See below example:
 
 ```ts
-// setup `stateManager` with some existing address
-const proof = await stateManager.getProof(address)
-const proofWithStorage = await stateManger.getProof(contractAddress, [storageKey1, storageKey2])
-
-const partialStateManager = await DefaultStateManager.fromProof(proof)
-// To add more proof data, use `addProofData`
-await partialStateManager.addProofData(proofWithStorage)
-const accountFromNewSM = await partialStateManager.getAccount(address)
-const accountFromOldSM = await stateManager.getAccount(address)
-console.log(accountFromNewSM, accountFromOldSM) // should match
-const slot1FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1)
-const slot2FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1) // should also match
+// ./examples/fromProofInstantiation.ts
+
+import { Account, Address, bytesToHex } from '@ethereumjs/util'
+import { DefaultStateManager } from '@ethereumjs/statemanager'
+import { hexToBytes } from '@ethereumjs/util'
+
+const main = async () => {
+  // setup `stateManager` with some existing address
+  const stateManager = new DefaultStateManager()
+  const contractAddress = new Address(hexToBytes('0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+  const byteCode = hexToBytes('0x67ffffffffffffffff600160006000fb')
+  const storageKey1 = hexToBytes(
+    '0x0000000000000000000000000000000000000000000000000000000000000001'
+  )
+  const storageKey2 = hexToBytes(
+    '0x0000000000000000000000000000000000000000000000000000000000000002'
+  )
+
+  await stateManager.putContractCode(contractAddress, byteCode)
+  await stateManager.putContractStorage(contractAddress, storageKey1, hexToBytes('0x01'))
+  await stateManager.putContractStorage(contractAddress, storageKey2, hexToBytes('0x02'))
+
+  const proof = await stateManager.getProof(contractAddress)
+  const proofWithStorage = await stateManager.getProof(contractAddress, [storageKey1, storageKey2])
+
+  const partialStateManager = await DefaultStateManager.fromProof(proof)
```

Comment:
(Inner check): if I would instantiate `fromProof(proofWithStorage)` then I should be able to retrieve the contract + 2 storage slots... right?

Reply:
I am currently checking this. The storage is present but the bytecode is not included in the proof so you cannot retrieve the contract code.